bat/ssh.go

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package bat
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"golang.org/x/crypto/ssh"
+)
+
+//SSHClient containes details of the server to be connect
+type SSHClient struct {
+	ClientIP   string
+	ClientPort string
+	ClientUser string
+	ClientPass string
+}
+
+//CreateSSHClient creates a SSH session for the server specified
+//
+//Receive a struct with the details of the server that will be connected
+//crete a tcp session that will be use to send and receive data
+//
+//sesion sould be created succesfully and wait for commands
+func CreateSSHClient(WLclient SSHClient) (*ssh.Client, error) {
+	//If no port is defined, it will be default to 22
+	if WLclient.ClientPort == "" {
+		WLclient.ClientPort = "22"
+	}
+	ClientFullAddress := fmt.Sprint(WLclient.ClientIP + ":" + WLclient.ClientPort)
+	config := &ssh.ClientConfig{
+		User: WLclient.ClientUser,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(WLclient.ClientPass),
+		},
+	}
+	Conection, err := ssh.Dial("tcp", ClientFullAddress, config)
+	return Conection, err
+}
+
+//SSHSendCommand  send a command to a specified server
+//
+//Recive a struct with the details of the server that will be connected
```

Comment:
Updated, I'll keep this in mind for the future, this time it help me to remember what the function should be doing